src/effect.js

```diff
@@ -0,0 +1,112 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+
+	p5.Effect = function(type) {
+		//everything points to the same audiocontext
+		this.type = type;
```

Comment:
Will all effects have `type` as the first argument in their constructor? `p5.Filter` already has this, but `p5.Reverb` and `p5.Delay` don't. I'd lean towards moving this to p5.Filter.